bitcoin/src/blockdata/transaction.rs

```diff
@@ -1168,6 +1170,69 @@ impl From<&Transaction> for Wtxid {
     fn from(tx: &Transaction) -> Wtxid { tx.wtxid() }
 }
 
+/// Predicts the weight of a to-be-constructed transaction.
+///
+/// This function computes the weight of a transaction which is not fully known. All that is needed
+/// is an arbitrary set of TxIn and TxOut.
+///
+/// # Arguments
+///
+/// * `inputs` - a vector of inputs (TxIn).
```

Comment:
Could this be achieved by just adding a new constructor to `InputWeightPrediction` which takes a `TxIn`?